bot/exts/utils/snekbox.py

```diff
@@ -217,6 +217,71 @@ async def send_eval(self, ctx: Context, code: str) -> Message:
             log.info(f"{ctx.author}'s job had a return code of {results['returncode']}")
         return response
 
+    @staticmethod
+    def get_time(results: dict) -> Tuple[str, Optional[str]]:
+        """
+        Parses time and output from results dict, if there was an error time is None and output contains the error.
+
+        Returns code output and timeit results
+        """
+        output = results["stdout"].strip("\n")
+
+        if results["returncode"] == 0:
+            time_index = output.rfind("\n")
+            if time_index != -1:
+                return output[:time_index], output[time_index:]
+            return "", output
+
+        return output, None
+
+    async def send_timeit(self, ctx: Context, code: str) -> Message:
+        """
+        Evaluate code with timing, format it, and send the output to the corresponding channel.
+
+        Return the bot response.
+        """
+        async with ctx.typing():
+            code = code.replace("'", r"\'")
+            code = f"import timeit\n" \
+                   f"timeit.main(['{code}'])"
+
+            results = await self.post_eval(code)
+            msg, error = self.get_results_message(results)
+            output, time = self.get_time(results)
+            icon = self.get_status_emoji(results)
+
+            if error:
+                output, paste_link = error, None
+            else:
+                output, paste_link = await self.format_output(output)
+
+            msg = f"{ctx.author.mention} {icon} {msg}.\n\n```\n{output}\n```"
+
+            if time:
+                msg = f"{msg}\nTime:\n```{time}```"
```

Comment:
The `output` will contain the time too if it's not truncated. You should remove it from the output.